test/e2e/e2e_test.go

```diff
@@ -0,0 +1,147 @@
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/xmudrii/etcdproxy-controller/pkg/apis/etcd/v1alpha1"
+	clientset "github.com/xmudrii/etcdproxy-controller/pkg/client/clientset/versioned"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// kubeconfigPath is path to the kubernetes configuration file.
+var kubeconfigPath = filepath.Join(os.Getenv("HOME"), "/.kube/config")
+
+// newKubeClient create new Kubernetes client instance based on kubeconfig file.
+func newKubeClient(kubeconfigPath string) (*kubernetes.Clientset, error) {
+	// use the current context in kubeconfig.
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// create the clientset.
+	return kubernetes.NewForConfig(config)
+}
+
+func newEtcdProxyClinet(kubeconfigPath string) (*clientset.Clientset, error) {
+	// use the current context in kubeconfig.
+	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+	if err != nil {
+		return nil, err
+	}
+
+	// create the clientset.
+	return clientset.NewForConfig(config)
+}
+
+func replicasetName(etcdstorageName string) string {
```

Comment:
My idea was to have a helper functions for calculating names, instead of hardcoding them in test cases, to make potential changes easier.

Reply:
> My idea was to have a helper functions for calculating names, instead of hardcoding them in test cases, to make potential changes easier.

I generally question the utility of one liners.  Do you think it's likely we'll change our minds?  How many places would we really have to update it?  *Should* such a change be hard since we're changing external behavior?

Reply:
> Do you think it's likely we'll change our minds? 

I don't think we'll ever change those. Even if we do in some case, that would be just a few lines to update. Let's get rid of those functions.